### DISCUSSION TITLE
Problem when installing Readarr on Homeassistant

### DISCUSSION THREAD

**Discussion by Trodfoot** [alexbelgium/hassio-addons#420] [General]
I thought since I am new at reporting errors I would first ask about it, rather than just put it in the issue area. Sorry if I am wrong.

So I am trying to install readarr as an addon to homeassistant and I go into configuration and change the location that put there /config/readarr  to /config/addons_config/readarr as that is where other addons are... When I go to the log to watch the container start up the config line that is passed to the program is being overwritten.  Here is part of the output:   [Info] AppFolderInfo: Data directory is being overridden to [/config/addons_config/radarr/readarr]    I didn't add the radarr I checked 3 times. 
The next line is:  [Info] MigrationController: *** Migrating data source=/config/addons_config/radarr/readarr/readarr. So it got worse.  I have the full log if its wanted,  I didn't know how to link it as a download.  At least the programmer had a sense of humour because the next line read: [Fatal] ConsoleApp: EPIC FAIL!, I just thought no kidding.

**Comment by alexbelgium** [comment]
Hi, Ahah, love programmer's humour :)

Actually it was a residual code issue in my addon, a line that was relevant at some point in the past but not anymore. I've pushed a new version that hopefully solves this. Thanks!

**Comment by Trodfoot** [comment]
Your fix worked like a charm thanks.  Another question about the same
addon. Should the addon not use the cifs username/password that i supply
when accessing the network drives i supply in the config?  Reason I ask is
when I go to add a root folder i get:

 Root Folder containing your book library
This must be different to the directory where your download client puts
files
Folder is not writable by user abc

It should not be using abc but the cifs user/pwd I supplied. abc was
probably used as a place holder while programming maybe?

**Comment by alexbelgium** [comment]
Abc is the user used in linuxserver io. You can assign a specific user is through the puid/pgid values in the addon options. So you should use as puid/pgid the values from your cifs user? Here is the documentation : https://docs.linuxserver.io/general/understanding-puid-and-pgid

**Comment by Trodfoot** [comment]
Hey Alex,

I guess I didn't explain good enough. I am getting that error even though I
specified a user to use in the options and both pgid/puid (changed both of
these to different ones, still no go). But instead of using the user I
supplied in cifsusername/cifspassword it is trying to use abc and whatever
password that was hardcoded in. It is not picking up my input name/password.

Where I am doing this is in Add Root Folder. I have tried it with multiple
directories  and 3 different Nas's. It has got me mixed up. I even did a
backup of my currently running readarr and restored it on your readarr. Now
I have all my book info and other settings but it still will not let me add
a root folder and says:

This must be different to the directory where your download client puts
files
Folder is not writable by user abc

And of course the log is complaining about the missing root folder i have
on the other machine. And it is different than the download clients
directory. I can keep the vm that readarr from linuxserver.io is running in
going, but was hoping to move it over under Home Assistant since I am
building it out.

**Comment by alexbelgium** [comment]
My understanding is that you need to use the same userid and groupeid as your cifs user in your puid pgid.

The best for you is to see the numerical id if the owner from your folders and use the same.

Or set both to 0 and be root

**Comment by alexbelgium** [comment]
Hi, this seems indeed to be as described here : https://askubuntu.com/questions/427107/why-can-i-create-users-with-the-same-uid
What is used to restrict access is not the username as such, but the user id/group id. By setting the same uid/gid for your "abc" user in linuxserver and your "normal" user in your other systems used to access the NAS, you can still access the same files.
